k8s/anthos/clusters.go

```diff
@@ -0,0 +1,59 @@
+package anthos
+
+import (
+	"context"
+
+	v1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// ClusterOps is an interface to perform k8s cluster operations
+type ClusterOps interface {
+	// ListCluster lists all kubernetes clusters
+	ListCluster(ctx context.Context, options metav1.ListOptions) (*v1beta1.ClusterList, error)
+	// GetCluster returns a cluster for the given name
+	GetCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.Cluster, error)
+	// DescribeCluster gets the cluster status
+	DescribeCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.ClusterStatus, error)
+}
+
+// ListCluster lists all kubernetes clusters
+func (c *Client) ListCluster(ctx context.Context, options metav1.ListOptions) (*v1beta1.ClusterList, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	clusterList := &v1beta1.ClusterList{}
+	err := c.RESTClient().Get().
+		Resource("clusters").
+		VersionedParams(&options, scheme.ParameterCodec).
+		Do(ctx).
+		Into(clusterList)
+
+	return clusterList, err
+}
+
+// GetCluster returns a cluster for the given name
+func (c *Client) GetCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.Cluster, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	cluster := &v1beta1.Cluster{}
+	err := c.RESTClient().Get().
+		Resource("clusters").
+		Name(name).
+		VersionedParams(&options, scheme.ParameterCodec).Do(ctx).Into(cluster)
+	return cluster, err
+}
+
+// DescribeCluster describe the cluster status
+func (c *Client) DescribeCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.ClusterStatus, error) {
```

Comment:
Can we change this function to something like `GetClusterStatus()` because it seems that what it returns and please have same comment on tops of the function like `// GetClusterStatus gets cluster and returns its status`, also add same description on tops on line 19

Reply:
Incorporated !!